modules/betsy/bc6h.glsl

```diff
@@ -0,0 +1,601 @@
+			block.z |= indices[4] << 29u;
+			block.w |= indices[5] << 0u;
+			block.w |= indices[6] << 3u;
+			block.w |= indices[7] << 6u;
+			block.w |= indices[8] << 9u;
+			block.w |= indices[9] << 11u;
+			block.w |= indices[10] << 14u;
+			block.w |= indices[11] << 17u;
+			block.w |= indices[12] << 20u;
+			block.w |= indices[13] << 23u;
+			block.w |= indices[14] << 26u;
+			block.w |= indices[15] << 29u;
+		}
+	}
+}
+
+layout(local_size_x = 8,
+		local_size_y = 8,
+		local_size_z = 1) in;
+
+void main() {
+	// gather texels for current 4x4 block
+	// 0 1 2 3
+	// 4 5 6 7
+	// 8 9 10 11
+	// 12 13 14 15
+	float2 uv = gl_GlobalInvocationID.xy * params.p_textureSizeRcp * 4.0f + params.p_textureSizeRcp;
+	float2 block0UV = uv;
+	float2 block1UV = uv + float2(2.0f * params.p_textureSizeRcp.x, 0.0f);
+	float2 block2UV = uv + float2(0.0f, 2.0f * params.p_textureSizeRcp.y);
+	float2 block3UV = uv + float2(2.0f * params.p_textureSizeRcp.x, 2.0f * params.p_textureSizeRcp.y);
+	float4 block0X = OGRE_GatherRed(srcTexture, pointSampler, block0UV);
+	float4 block1X = OGRE_GatherRed(srcTexture, pointSampler, block1UV);
+	float4 block2X = OGRE_GatherRed(srcTexture, pointSampler, block2UV);
+	float4 block3X = OGRE_GatherRed(srcTexture, pointSampler, block3UV);
+	float4 block0Y = OGRE_GatherGreen(srcTexture, pointSampler, block0UV);
+	float4 block1Y = OGRE_GatherGreen(srcTexture, pointSampler, block1UV);
+	float4 block2Y = OGRE_GatherGreen(srcTexture, pointSampler, block2UV);
+	float4 block3Y = OGRE_GatherGreen(srcTexture, pointSampler, block3UV);
+	float4 block0Z = OGRE_GatherBlue(srcTexture, pointSampler, block0UV);
+	float4 block1Z = OGRE_GatherBlue(srcTexture, pointSampler, block1UV);
+	float4 block2Z = OGRE_GatherBlue(srcTexture, pointSampler, block2UV);
+	float4 block3Z = OGRE_GatherBlue(srcTexture, pointSampler, block3UV);
+
+	float3 texels[16];
+	texels[0] = float3(block0X.w, block0Y.w, block0Z.w);
+	texels[1] = float3(block0X.z, block0Y.z, block0Z.z);
+	texels[2] = float3(block1X.w, block1Y.w, block1Z.w);
+	texels[3] = float3(block1X.z, block1Y.z, block1Z.z);
+	texels[4] = float3(block0X.x, block0Y.x, block0Z.x);
+	texels[5] = float3(block0X.y, block0Y.y, block0Z.y);
+	texels[6] = float3(block1X.x, block1Y.x, block1Z.x);
+	texels[7] = float3(block1X.y, block1Y.y, block1Z.y);
+	texels[8] = float3(block2X.w, block2Y.w, block2Z.w);
+	texels[9] = float3(block2X.z, block2Y.z, block2Z.z);
+	texels[10] = float3(block3X.w, block3Y.w, block3Z.w);
+	texels[11] = float3(block3X.z, block3Y.z, block3Z.z);
+	texels[12] = float3(block2X.x, block2Y.x, block2Z.x);
+	texels[13] = float3(block2X.y, block2Y.y, block2Z.y);
+	texels[14] = float3(block3X.x, block3Y.x, block3Z.x);
+	texels[15] = float3(block3X.y, block3Y.y, block3Z.y);
+
+	uint4 block = uint4(0u, 0u, 0u, 0u);
+	float blockMSLE = 0.0f;
+
+	EncodeP1(block, blockMSLE, texels);
+#ifdef QUALITY
+	for (uint i = 0u; i < 32u; ++i) {
+		EncodeP2Pattern(block, blockMSLE, i, texels);
+	}
+#endif
```

Comment:
```suggestion
	EncodeP1(block, blockMSLE, texels);
#ifdef QUALITY
	for (uint i = 0u; i < 64u; ++i) {
		EncodeP2Pattern(block, blockMSLE, i, texels);
	}
#endif
```

I wonder if I'll cause a driver hang if I double the search space quality. What's your setup to output the three images for bc6?

Reply:
I'm using this image as a source: [testout.zip](https://github.com/godotengine/godot/files/15274278/testout.zip).

For CVTT I disable the GPU compressor from the settings, for Betsy I leave it enabled, and for DirectXTex I use a custom program to convert the EXR to DDS with BC6SF compression.

Reply:
I'll stop poking at it, but one approach is to use more block modes but I have to go now.